specs/agents/log-sending.md

```diff
@@ -0,0 +1,59 @@
+# Log sending
```

Comment:
(These are ambiguous questions that may be out of scope for this PR.)

What happens if lots of log records/events in some user apps leads to hitting the agent's max queue size for sending events to APM server more frequently? Would we notice? Should [apm-agent health metrics](https://github.com/elastic/apm/pull/708) events have a higher priority?

Should there be some way to reduce the number of sent logs if it is clear they are dominating and overloading intake usage?  Something like `transaction_sample_rate` -- but obviously would have to be different because a random sampling of log messages would likely make them useless. Perhaps some maximum number of log records/events per time period (say 100 per intake batch or streaming request period, or some value that would avoid a heavy-logging app immediately overloading the agent's event queue). Perhaps start off with inifity for this value and have automatic backoff if the agent's event queue is getting filled from log events. I am just thinking out loud. I haven't spent any time thinking this through.

Reply:
Sorry for the late reply, I somehow missed the notification on this one.

> I think the Java implementation of health metrics includes a breakdown of sent/dropped events per event type, so any "logs overflow" situation should be visible in those health metrics at the condition that the agent is able to send them (adding distinct queues would solve this aspect), can you confirm @JonasKunz ?

The metric for "total number of events created" does have a label for the event type, so we should be able to see if the number of logs created exceeds the number of other events significantly.
We however do not breakdown the "number of dropped events" by event type. This is because when the APM server rejects events (e.g. due to validation), it does not tell the agent which events or event types were rejected.